packages/vsx-registry/src/browser/vsx-extension.tsx

```diff
@@ -100,6 +103,25 @@ export class VSXExtension implements VSXExtensionData, TreeElement {
 
     protected readonly data: Partial<VSXExtensionData> = {};
 
+    protected _width: number;
+    get width(): number {
+        return this._width;
+    }
+    set width(width: number) {
+        this._width = width;
```

Comment:
On resitze event, you should call `VSXExtensionEditorComponent.setState` and keep in the state everything what is required for rendering. `render` function should not compute any new properties but simply use state and props to render. That's how react works.